book/src/integrations.md

```diff
@@ -44,6 +44,36 @@ In case of a run-time error occurring during templating, the response will be of
 signature, with a status code of `500 Internal Server Error`, mime `*/*`, and an empty `Body`.
 This preserves the response chain if any custom error handling needs to occur.
 
+## Hyper integration
+
+Enabling the `with-hyper` feature appends `From<Template> for hyper::Response<hyper::Body>` and
+`TryFrom<Template> for hyper::Body` implementations for each template type. These
+provide the ability for Hyper apps to build a response directly from
+a template, or to create a templated hyper body to be used further in building a
+`Response`. See [the example](https://github.com/djc/askama/blob/main/askama_hyper/tests/basic.rs)
+from the Askama test suite for more on how to integrate.
+
+When using `Template::from()` (or `Template::into()`),
+the returned type is `hyper::Response<hyper::Body>`.
+On the success of run-time rendering, the response consists of the status code `200`, the MIME type associated with the
+template, and the body, created from the rendered String.
+On the failure of run-time rendering, the response consists of the status code `500`, and an empty
+body. The returned value can be seen being created in the integration's public `respond` function
+in the
+[integration's source](https://github.com/djc/askama/blob/main/askama_hyper/src/lib.rs).
+
+When using `Template::try_from()` (or `Template::try_into()`),
+the returned type is `Result<hyper::Body, askama::Error>`.
+The generated code for the template does much less in this case, returning either a `hyper::Body`
+when rendering was successful or the `askama::Error` when it was not. This is meant to allow the
+caller more flexibility in how a response is built and in actions taken when rendering fails.
+
+The
+[integration's source](https://github.com/djc/askama/blob/main/askama_hyper/src/lib.rs)
+also provides a public `try_respond` function that some may find useful to work with or work from.
+It should not be confused with the `TryFrom` generated impl however; the generated impl
+does not use either of the functions found in that source.
```

Comment:
What do you mean with this paragraph? The generated TryFrom implementation does indeed use this function: https://github.com/djc/askama/blob/1856c4b57ca7ab706552e2c47f550c86e8999e7c/askama_derive/src/generator.rs#L461

Reply:
This what what I thought the generated `TryFrom` function was. Am I misinterpreting something?
```rust
impl < 'a, N > ::core::convert::TryFrom <&HelloTemplate<'a, N>> for ::askama_hyper::hyper::Body
where N: std::fmt::Display,
{
    type Error = ::askama::Error;
    #[inline]
    fn try_from(value : & HelloTemplate < 'a, N >) -> Result < Self, Self :: Error> {
        ::askama::Template::render(value).map(Into::into)
    }
}
```
This looks like it just calls `render`, gets a `String`, and has the compiler convert that to a `hyper::Body`. I didn't think it uses `respond` or `try_respond`. I'll admit the rust compiler still has me baffled sometimes. I thought I had this one figured out though. Apologies. You know the crate better than I. Please suggest mods or make the necessary cleanups.

Reply:
Ah, yes, the `<Body as TryFrom<Template>>` does not use `try_respond`, only `<Response<Body> as TryFrom<Template>>` does:
`impl From<Template> for hyper::Body` → `askama_hyper::respond()` → `askama_hyper::try_respond()`.

I would call this an implementation detail which may change at any point, though. Do you think it should be highlighted to the user?

Reply:
Really? First I'm wrong. Then the public API has details that don't need to be documented because they may change at any point? I think something else is going on here. I won't push this further.

Reply:
I guess you did not express myself properly. I only meant that the user should not need to know (or care) how `TryFrom<Template> for hyper::Body` was implemented. That the functions exists is not an implementation detail.

And I only commented on the third paragraph. I think that documenting the hyper integration is a good addition to the book, not only for new users of Askama, but all users.